Implement undo and redo

penbuvt opened the pull request:
Implement undo and redo functionality and expose it in the UI through Undo and Redo buttons. Accesskeys `u` and `r` have been added as poor-man's keybinds.

To do this, a new `ThemeHistory` module was added that contains two stacks, one for undo and another for redo. The `Theme` class instantiates a copy, exposes it through the `history` property, and implements the `commit`, `undo` and `redo` methods for managing `this.items`. The `WinClassicTheme` overrides these methods to also update the UI.

Resolves #6

<img width="236" height="84" alt="image" src="https://github.com/user-attachments/assets/8048acfb-bab5-4ccb-a603-4d6e4c32622a" />

penbuvt committed:
Add ThemeHistory class

This currently has functions to commit and undo from an array-based
stack.

penbuvt committed:
Commit colour input changes to history stack

Introduce new history property to Theme and have the WinClassic class
commit on colour picker change, as well as app initialization.

penbuvt committed:
Deep clone items added to history

Items must be deep cloned at the boundary of the history module to
prevent accidental modifications to history items.

penbuvt committed:
Add undo functionality to UI

Add an undo button in the UI and override the inherited undo method in
WinClassicTheme to also update the UI.

penbuvt committed:
Disable Undo button at first undo state

When there is nothing to undo, communicate that to the user by disabling
the undo button.

penbuvt committed:
Support redo functionality

Add a new redo queue to the theme history module to keep track of redo
states.

penbuvt committed:
Separate current state from undo stack

Make the history module's undo stack more intuitive by separating out
the current state. Previously, the current state was at the top stack.
But now the undo stack is strictly for past states and the current state
is in its own variable.

penbuvt committed:
Add accesskeys to Undo and Redo buttons

Poor-man's keybinds for undo and redo: accesskey+u and accesskey+r
respectively.

tpenguinltg commented on scripts/theme-history.js:
```diff
@@ -0,0 +1,45 @@
+}
+
+ThemeHistory.prototype.redo = function() {
+  this.undoHistory.push(this.current)
```
Missing semicolon
```suggestion
  this.undoHistory.push(this.current);
```

tpenguinltg commented on scripts/winclassic.js:
```diff
@@ -111,6 +119,11 @@ WinClassicTheme.prototype.onColorChange = function(e) {
   this.resetPickers();
 }
 
+WinClassicTheme.prototype.onColorCommit = function(e) {
```
Unused parameter
```suggestion
WinClassicTheme.prototype.onColorCommit = function() {
```

tpenguinltg reviewed (approved):
Some very minor issues, but looks good to me otherwise!

tpenguinltg committed:
Clean up minor code infractions

- Missing semicolon in ThemeHistory.prototype.redo
- Unused parameter in WinClassicTheme.prototype.onColorCommit
